odbc/src/api/functions.rs

```diff
@@ -1278,27 +1279,47 @@ pub unsafe extern "C" fn SQLGetData(
     let mut ret = Bson::Null;
     let mongo_handle = MongoHandleRef::from(statement_handle);
     {
```

Comment:
Another parameter that we can check is the buffer_length.  Looks like a any value less than 0 for BufferLength is invalid and should set the sqlstate to `HY090`
```
HY090 | Invalid string or buffer length | (DM) The value specified for argument BufferLength was less than 0.
```

Reply:
Also set by the DM.